Move from Alpine to Ubuntu

Seems to be the quickest way to get #37 working.

As per our phone call Friday I tried fixing the evanescent build to **not** include gnu libc but unfortunately it didn't work out.

Most of the issues were solved (i.e. most undefined references were gone) when runtime-linking to the alpine gnu libraries, but not all. For example, libssl still had an undefined reference to `__isoc23_strtol` and that seems to be a gnu-libc option that Alpine does not give us in the compatibility library.

See for example https://github.com/openssl/openssl/issues/22833

So I think that the easiest is that we do move to Ubuntu:-(

--- a/Dockerfile
+++ b/Dockerfile
@@ -6,12 +6,10 @@ ADD ./package[s] /packages
 WORKDIR /code
 
 # Add evanescent directory to library search path
-RUN echo "/packages/evanescent" > /etc/ld.so.conf.d/evanescent.conf
-RUN ldconfig
+RUN echo "/packages/evanescent" > /etc/ld.so.conf.d/evanescent.conf && ldconfig
 
 # Install Python for TCP reflector
-RUN apt update
-RUN apt install -y python3
+RUN apt update && apt install -y python3
 
 RUN yarn install && \
     yarn build && \

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,16 +1,13 @@
-FROM node:22-alpine3.18
+FROM node:22
 
 ADD . /code/
 ADD ./package[s] /packages
 
 WORKDIR /code
 
-# These packages are needed to run evanescent and the webrtc sfu
-RUN apk add gcompat
-# Add evanescent directory to search path
-RUN echo "/lib:/usr/local/lib:/usr/lib:/packages/evanescent" > /etc/ld-musl-x86_64.path
 # Install Python for TCP reflector
-RUN apk add --update --no-cache python3
+RUN apt update
+RUN apt install -y python3
 
 RUN yarn install && \
     yarn build && \

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,6 +7,7 @@ WORKDIR /code
 
 # Add evanescent directory to library search path
 RUN echo "/packages/evanescent" > /etc/ld.so.conf.d/evanescent.conf
+RUN ldconfig
 
 # Install Python for TCP reflector
 RUN apt update

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM node:22
+FROM node:22-bookworm-slim
 
 ADD . /code/
 ADD ./package[s] /packages

--- a/Dockerfile
+++ b/Dockerfile
@@ -5,6 +5,9 @@ ADD ./package[s] /packages
 
 WORKDIR /code
 
+# Add evanescent directory to library search path
+RUN echo "/packages/evanescent" > /etc/ld.so.conf.d/evanescent.conf
+
 # Install Python for TCP reflector
 RUN apt update
 RUN apt install -y python3

